src/include/platform/CHIPDeviceConfig.h

```diff
@@ -955,6 +955,16 @@
  *   The value is base-64 encoded string.
  */
 #ifndef CHIP_DEVICE_CONFIG_USE_TEST_SPAKE2P_VERIFIER
+
+#if CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE != 20202021
+#error "Non-default Spake2+ passcode configured but verifier left unchanged"
```

Comment:
What will happen if none of the CHIP_DEVICE_CONFIG_USE_TEST* bits are defined at all?  Will that `#if` test true of false?

Reply:
If they are not defined by a user, they are set to defaults right above, so they are always defined at this place.

Reply:
This works in a case where the setup pin code is not provided in tthe nvm key else `CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE` is unchanged but not used.

```
err = mGenericConfigManager.ReadConfigValue(ConfigClass::kConfigKey_SetupPinCode, setupPasscode);
#if defined(CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE) && CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE
    if (err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
    {
        setupPasscode = CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE;
        err           = CHIP_NO_ERROR;
    }
#endif // defined(CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE) && CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE
    SuccessOrExit(err);

```

Reply:
Yes, but if someone stores the passcode in the nvm or elsewhere, is there a reason that they keep `CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE` defined? I simply wanted to help people who simply play around with examples in the repo. I assume that if someone is familiar with the code enough to provide their own implementation or nvm data, they will know about the verifier, right?